syntax = "proto2";

package Codec_NodeStatus;

message NodeChainState
{
	required uint32 state = 1;
	optional float blockSpeed = 2;
	optional uint32 eta = 3;
	optional float pct = 4;
	optional uint32 blocksLeft = 5;
}

message NodeStatus
{
	required uint32 status = 1;
	required bool SegWitEnabled = 2;
	required uint32 rpcStatus = 3;
	optional NodeChainState chainState = 4;
}

message ProgressData
{
	required uint32 phase = 1;
	optional double progress = 2;
	optional fixed32 time = 3;
	optional uint32 numericProgress = 4;
	repeated bytes ID = 5;
}

message BDV_Error
{
	required uint32 type = 1;
	optional string error = 2;
	optional string extra = 3;
}

message Refresh
{
	required uint32 refreshType = 1;
	repeated bytes id = 2;
}